### ISSUE TITLE
4K Movies not Filtered when "HD" Video Type is Selected

### ISSUE THREAD

**Issue by MrPeteypoo** [jellyfin/jellyfin-web#6067]
**Describe The Bug**
When using the Jellyfin web interface or the desktop media player on macOS, I noticed a problem with the "Video Types" filter. My library is made up of rips of my personal DVDs, Blu-rays and UHD Blu-rays which have been remuxed into MKV files. This is what happens when I select the "SD", "HD" and "4K" filters:
- SD: 7 / 101 movies are visible, all of which are my DVD rips
- 4K: 70 / 101 movies are visible, all of which are my UHD Blu-ray rips
- HD: 94 / 101 movies are visible, both my UHD Blu-ray and FHD Blu-ray rips are shown

This is unintuitive to me because I have a means of only seeing my DVDs, and only seeing my UHD Blu-rays, but I don't have a way of only seeing my 1080p Blu-rays.

**Steps To Reproduce**
1. Create a movie library
2. Add an MKV file with a resolution of 3840x2160
3. Add an MKV file with a resolution of 1920x1080
4. Add an MKV file with a resolution of 720x480
5. Go to the movie library
6. Click on the filter button
7. Click on "Video Types"
8. Click on "HD"
9. Notice how both the 1080p and 2160p videos are visible

**Expected Behavior**
Given my library of movies I would expect the following outcomes:
- SD: 7 / 101 movies are visible, same as the current behaviour
- 4K: 70 / 101 movies are visible, same as the current behaviour
- HD: 24 / 101 movies are visible, all of which are my FHD Blu-ray rips

**Logs**
N/A

**Screenshots**
![image](https://github.com/user-attachments/assets/fa303864-9a85-412b-ba53-2c5aef6eb6fd)

**System (please complete the following information):**
 - Platform: macOS 14.6.1
 - Browser: Brave (latest version)
 - Jellyfin Version: 10.9.10

**Additional Context**
I hope you agree with me that the current behaviour is incorrect, I can see why people might want 4K movies to display when they select a "HD" video type because 4K is often referred to as a high definition format but if the current behaviour is maintained then I would appreciate a "FHD" video type as that would be very helpful.

**Comment by mihawk90** [comment]
This is interesting, could you by chance test the 4K filter again? From what I can tell it isn't filtering anything for me at all, it just shows the full library. So this behaviour might have changed at some point.

**Comment by MrPeteypoo** [comment]
> This is interesting, could you by chance test the 4K filter again? From what I can tell it isn't filtering anything for me at all, it just shows the full library (and someone else on Matrix tested with the same result). So this behaviour might have changed at some point.

Yes this bug has evolved since I first made it. Now only the SD filter works for me. 4K shows 1080p movies now. I'm running Jellyfin 10.10.3 or whatever the latest is.

**Comment by mihawk90** [comment]
Probably worth changing the issue title then, "HD and 4K filters non-functional" or something along those lines.

**Comment by MrPeteypoo** [comment]
At the risk of seeming overly dramatic I'm going to abandon this because I don't want to maintain this bug report and update it based on new server/web versions. Hopefully the filters get some love in the future but even if they do I don't really want to champion whether "HD" should exclude 4K movies or not. Feel free to make a new issue about the general broken filters if you'd like to.

**Comment by Tech180** [comment]
I just wanted to add that I am currently experiencing this issue. I agree that it isn't a huge deal, but still something to note for the developers. I am unsure why only the SD filter works, but I hope they see this as a potential issue.

**Comment by mihawk90** [comment]
> However, I was unable to reproduce the new issue where the filters aren't working

Might be a misunderstanding here. I don't think "not working" is the correct term, more of an unexpected behaviour.
I don't have any SD content to test with but going by the comments above, the SD filter works as expected. It's just the HD filters showing both (Full)HD/1080p content as well as UHD/4K/2160p content.
What I have not looked into is how HD/720p content behaves (or is supposed to behave). From a TV-Marketing perspective it used to be marketed as "HD ready" I believe, but whether one considers this HD today is up to the user I guess.

There is one other oddity but I think that's more of a jellyfin-web problem:
The HD and SD filter are mutually exclusive, i.e. users can only select one at a time. But the 4K filter is additive, i.e. it can be selected in combination with SD ***or*** HD (or standalone).
These should probably all be separate independent filters. What if a user wants to show both HD and SD content, but no 4K or BD/DVD/3D content? That doesn't seem possible at the moment.

**Comment by theguymadmax** [comment]
Ok, thanks. Here are a few more details for anyone interested. 

How the filters currently work:

HD:
* If query.IsHD = true, the server sets the minWidth to 1200px.  (Since there is no maxWidth "4k" resolutions will be picked up)
* If query.IsHD = false, the server sets the maxWidth to less than 1200px.

4K:
* If query.Is4K = true, the server sets the minWidth to 3800px.
* If query.Is4K = false, the server sets the minWidth to less than 3800px.

SD:
* SD is not explicitly queried on the server. Instead, it is inferred from the absence of the IsHD filter.
* This is determined in the web application by checking if isHd = false.
* As you mentioned, this is mutually exclusive: that's why you can select HD and SD at the same time, but can't select both 4K and SD.

	
Two issues with the current code, that I'm addressing with my PR:

1. The HD filter does not exclude 4K content because there is no maxWidth. As a result, 4K content is included when the HD filter is applied.

2. The 4K filter sets a minWidth of 3800 to be considered 4K. This can lead to many high-resolution files being classified as HD, as the 3800 threshold may be too high for some files that should be considered 4K. See the [discussion](https://github.com/jellyfin/jellyfin-meta/discussions/86) on adjusting this threshold.

**Comment by mihawk90** [comment]
> * that's why you can select HD and SD at the same time, but can't select both 4K and SD.

I think you meant HD+4K, or SD+4K at the same time? Selecting HD unselects SD and vice versa.
But regardless, as mentioned IMO these should all be separate filters that only select their respective resolutions.
I.e. SD only selects SD content, 4K only 4K content, and HD everything in between (although it might be worth looking into "HD ready" and "FullHD" at the same time).
Being able to combine them freely is IMO the expected behaviour because otherwise what's the point of having them? Users *should* be able to combine the filters as needed.

The current selection is made even worse by the fact the selectors have no visual distinction between them, so it actually looks like a bug. The selectors are Checkboxes, and most of them behave that way. But the SD and HD filters behave like Radio Buttons instead and aren't even next to each other.

**Comment by MrPeteypoo** [comment]
Alright I'll chime in again and since I have more free time I'll reopen the issue. As of Jellyfin 10.10.4 this is my experience.

Context:
In my "Films" library I have a mixture of DVD, Blu-ray and UHD blu-ray rips which are remuxed in an MKV format. Some movies exist in collections where I deem appropriate. I have roughly 400 movies in total.

Current Behaviour:
- No filtering applied: I see a library of 243 entries with many movies hidden behind collections, this is expected and intended
- SD filter applied: I see 42 entries, collections have been expanded and every movie displayed is a DVD remux. This is desired/expected.
- HD filter applied: I see 362 entries, collections have been expanded but every movie displayed is either a blu-ray (FHD) or UHD blu-ray (4K) remux. This is not desired, the UHD movies should not be visible
- 4K filter applied: I see my entire library of 243 entries, collections HAVE NOT been expanded, the filter doesn't do anything at all. This is not desired, I expect my collections to be expanded and only UHD movies should be visible.
